src/Elastic.Apm/Metrics/Windows/GlobalMemoryStatusEx.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Runtime.InteropServices;
+
+namespace Elastic.Apm.Metrics.Windows
+{
+	[StructLayout(LayoutKind.Sequential)]
+	internal struct MemoryStatusEx
+	{
+		internal uint dwLength;
+		private readonly uint dwMemoryLoad;
+		internal readonly ulong ullTotalPhys;
+		internal readonly ulong ullAvailPhys;
+		private readonly ulong ullTotalPageFile;
+		private readonly ulong ullAvailPageFile;
+		private readonly ulong ullTotalVirtual;
+		private readonly ulong ullAvailVirtual;
+		private readonly ulong ullAvailExtendedVirtual;
+	}
+
+	internal static class GlobalMemoryStatus
+	{
+		[return: MarshalAs(UnmanagedType.Bool)]
+		[DllImport("Kernel32.dll", CharSet = CharSet.Auto, SetLastError = true)]
+		private static extern bool GlobalMemoryStatusEx(ref MemoryStatusEx lpBuffer
+		); //ref. vs. [In, Out], see: https://www.pinvoke.net/default.aspx/kernel32.globalmemorystatusex
+
+		internal static (bool success, ulong total, ulong avail) GetTotalPhysAndAvailPhys()
```

Comment:
I think if it's a library method it should use a named struct for the return value type.

Reply:
Can't follow. What do you mean by library method? This is an internal method within the agent. Or you mean we should return `MemoryStatusEx`?

Reply:
I mean it's an util method that we might want to use in another context (not some private method accessible only to one class). It's very easy to make and deconstruct the tuple in the wrong order. With named struct/class for return value those types of bugs are not possible. I think returning `MemoryStatusEx` unnecessarily exposes implementation detail so I would just declare a dedicated struct/class (`MemoryStatus`) and return it.

Reply:
> It's very easy to introduce a bug by deconstructing the tuple in a wrong order. 

These are named tuples... the compiler knows those names, that language feature is designed to address your issue.
<img width="466" alt="image" src="https://user-images.githubusercontent.com/1091853/59044062-b32ef780-887d-11e9-8a24-90c4a5c371ba.png">

I think this code is fine.

Reply:
Not if you deconstruct the tuple, which is the way this method's return value is actually used in the code  (`FreeAndTotalMemoryProvider.GetSamples`). If you replace

`var (success, totalMemory, freeMemory) = GlobalMemoryStatus.GetTotalPhysAndAvailPhys();`

by 

`var (success, freeMemory, totalMemory) = GlobalMemoryStatus.GetTotalPhysAndAvailPhys();`

It will compile and wait for the support case.